lib/inspec/resources/host.rb

```diff
@@ -296,15 +306,44 @@ def ping(hostname, port = nil, _proto = nil)
     end
 
     def resolve(hostname)
+      addresses = []
+      # -Type A is the DNS query for IPv4 server Address.
       cmd = inspec.command("Resolve-DnsName –Type A #{hostname} | ConvertTo-Json")
       begin
-        resolv = JSON.parse(cmd.stdout)
+        resolve_ipv4 = JSON.parse(cmd.stdout)
       rescue JSON::ParserError => _e
         return nil
       end
 
-      resolv = [resolv] unless resolv.is_a?(Array)
-      resolv.map { |entry| entry["IPAddress"] }
+      resolve_ipv4 = resolve_ipv4.inject(:merge) if resolve_ipv4.is_a?(Array)
```

Comment:
@ahasunos Hi there; It looks like this breaks functionality when there is more than one IPAdddress present

i.e. Consider the following result from powershell:

```powershell
PS > Resolve-DnsName -Type A dc1.example.demo.local | Select-Object -first 2 | ConvertTo-Json
[
    {
        "IP4Address":  "10.10.11.5",
        "Name":  "dc1.example.demo.locall",
        "Type":  1,
        "CharacterSet":  1,
        "Section":  0,
        "DataLength":  4,
        "TTL":  1200,
        "Address":  "10.10.11.5",
        "IPAddress":  "10.10.11.5",
        "QueryType":  1
    },
    {
        "IP4Address":  "10.0.2.15",
        "Name":  "dc1.example.demo.local",
        "Type":  1,
        "CharacterSet":  1,
        "Section":  0,
        "DataLength":  4,
        "TTL":  1200,
        "Address":  "10.0.2.15",
        "IPAddress":  "10.0.2.15",
        "QueryType":  1
    }
]
```

After `resolve_ipv4.inject(:merge)` this would become:

``` ruby                                                                                         
{"IP4Address"=>"10.0.2.15",                                                                           
 "Name"=>"dc1.example.demo.local",                                                                       
 "Type"=>1,                                                                                           
 "CharacterSet"=>1,                                                                                   
 "Section"=>0,
 "DataLength"=>4,
 "TTL"=>1200,
 "Address"=>"10.0.2.15",
 "IPAddress"=>"10.0.2.15",
 "QueryType"=>1}
```

Which causes the the `10.10.11.5` IPAddress to be dropped; If I understand correctly the previous implementation before this change did not have this bug - https://github.com/inspec/inspec/blame/172ab5e49a360e18d04eff42de7c461cc3d15069/lib/inspec/resources/host.rb#L306-L307

Reply:
Thanks @AlanFoster 
Let me look into it.